Emit -points does not add time

bkutil opened the pull request.

bkutil commented:
cc: @dmehra @go-oleg @dmajda

bkutil commented:
refs: #77

dmehra commented:
Does this also cover "not emitting ticks and emitting all points at once" as per last comments in #77?

bkutil commented:
Yes, updated the commit message.

demmer commented:
This looks good, though I think we want to add a check that the user shouldn't use `-points` with any of `-from, -to, -last, -limit, -every` since they'll now be ignored.

demmer commented:
Also given this new behavior, you should modify `next_point` so it doesn't have the check for `this.points` since that'll never happen any more.

bkutil commented:
> This looks good, though I think we want to add a check that the user shouldn't use -points with any of -from, -to, -last, -limit, -every since they'll now be ignored.

Ok, I added the missing checks and fixed the tests.

> Also given this new behavior, you should modify next_point so it doesn't have the check for this.points since that'll never happen any more.

Won't `this.points` check still be needed when we iterate over `-points` containiing time?

demmer commented on lib/runtime/procs/emit.js:
```diff
@@ -54,23 +54,30 @@ var emit = source.extend({
             if (_.has(options, 'to') || _.has(options, 'limit')) {
                 throw this.compile_error('RT-EMIT-POINTS-LIMIT-ERROR');
             }
+            if (_.has(options, 'from') || _.has(options, 'every')) {
```
What about `-to`?

bkutil replied:
Three lines above this one? :)

demmer commented:
> Won't this.points check still be needed when we iterate over -points containing time?

Oh of course -- I misread the PR. You're correct.

Modulo the missing check for `-to` I think this looks good. We will have a conflict once #248 merges since I reworked a bit of the internals of `emit`, but that should be easy enough to resolve.
